packages/venia-ui/lib/components/CheckoutPage/OrderConfirmationPage/orderConfirmationPage.js

```diff
@@ -1,34 +1,121 @@
-import React from 'react';
+import React, { useEffect } from 'react';
+import { propType } from 'graphql-anywhere';
+import { number, shape, string } from 'prop-types';
+import { useOrderConfirmationPage } from '@magento/peregrine/lib/talons/CheckoutPage/OrderConfirmationPage/useOrderConfirmationPage';
 
-import OrderSummary from './orderSummary';
+import { mergeClasses } from '../../../classify';
+import { Title } from '../../../components/Head';
 import CreateAccount from './createAccount';
 import ItemsReview from '../ItemsReview';
-import Subscribe from './subscribe';
+import defaultClasses from './orderConfirmationPage.css';
 
-import classes from './orderConfirmationPage.css';
+import { GET_ORDER_DETAILS } from '../checkoutPage.gql';
+
+const OrderConfirmationPage = props => {
+    const classes = mergeClasses(defaultClasses, props.classes);
+    const { data, orderNumber } = props;
+
+    const talonProps = useOrderConfirmationPage({
+        data
+    });
+
+    const { flatData, isSignedIn } = talonProps;
+
+    const {
+        city,
+        country,
+        email,
+        firstname,
+        lastname,
+        postcode,
+        region,
+        shippingMethod,
+        street
+    } = flatData;
+
+    const streetRows = street.map((row, index) => {
+        return (
+            <span key={index} className={classes.addressStreet}>
+                {row}
+            </span>
+        );
+    });
+
+    useEffect(() => {
+        window.scrollTo({
+            left: 0,
+            top: 0,
+            behavior: 'smooth'
+        });
+    }, []);
+
```

Comment:
@sirugh — Just submitted a PR that avoids needing code to scroll to top in every module on route change. See #2309

Reply:
Awesome! I'll leave this here for now and we can go clean up extra stuff later depending on which PR gets merged first :) I will say that this isn't a route, so it may not actually trigger the effect.